spec/turn_spec.rb

```diff
@@ -0,0 +1,375 @@
+        card8 = Card.new(:diamond, '2', 2)
+
+        
+
+        deck1 = Deck.new([card1, card2, card5, card8])
+        deck2 = Deck.new([card3, card4, card6, card7])
+
+
+        player1 = Player.new("Megan", deck1)
+        player2 = Player.new("Aurora", deck2)
+
+        turn = Turn.new(player1, player2)
+
+        expect(player1.deck.rank_of_card_at(0)).to eq (player2.deck.rank_of_card_at(0))
+
+        expect(player1.deck.rank_of_card_at(2)).to eq (player2.deck.rank_of_card_at(2))
+
+        expect(turn.type).to eq(:mutually_assured_destruction)
+
+    end
+
+    it "determines the winner if turn type is ':basic'" do
+
+        card1 = Card.new(:heart, 'Jack', 11)
+        card2 = Card.new(:heart, '10', 10)
+        card3 = Card.new(:heart, '9', 9)
+        card4 = Card.new(:diamond, 'Jack', 11)
+        card5 = Card.new(:heart, '8', 8)
+        card6 = Card.new(:diamond, 'Queen', 12)
+        card7 = Card.new(:heart, '3', 3)
+        card8 = Card.new(:diamond, '2', 2)
+
+        
+
+        deck1 = Deck.new([card1, card2, card5, card8])
+        deck2 = Deck.new([card3, card4, card6, card7])
+
+
+        player1 = Player.new("Megan", deck1)
+        player2 = Player.new("Aurora", deck2)
+
+        turn = Turn.new(player1, player2)
+
+        expect(turn.type).to eq(:basic)
+
+        expect(turn.winner).to eq(player1)
+
+    end
+
+    it "determines winner if type of turn is ':war'" do
+
+        card1 = Card.new(:heart, '10', 10)
+        card2 = Card.new(:spade, '10', 10)
+        card3 = Card.new(:diamond, '10', 10)
+        card4 = Card.new(:diamond, 'Jack', 11)
+        card5 = Card.new(:heart, '8', 8)
+        card6 = Card.new(:diamond, 'Queen', 12)
+        card7 = Card.new(:heart, '3', 3)
+        card8 = Card.new(:diamond, '2', 2)
+
+        
+
+        deck1 = Deck.new([card1, card2, card5, card8])
+        deck2 = Deck.new([card3, card4, card6, card7])
+
+
+        player1 = Player.new("Megan", deck1)
+        player2 = Player.new("Aurora", deck2)
+
+        turn = Turn.new(player1, player2)
+
+        expect(turn.type).to eq(:war)
+
+        expect(turn.winner).to eq(player2)
+
+    end
+
+    it "determines draw if turn type is 'mutually_assured_destruction" do
+
+        card1 = Card.new(:heart, '10', 10)
+        card2 = Card.new(:spade, '10', 10)
+        card3 = Card.new(:diamond, '10', 10)
+        card4 = Card.new(:diamond, 'Jack', 11)
+        card5 = Card.new(:heart, 'Queen', 12)
+        card6 = Card.new(:diamond, 'Queen', 12)
+        card7 = Card.new(:heart, '3', 3)
+        card8 = Card.new(:diamond, '2', 2)
+
+        
+
+        deck1 = Deck.new([card1, card2, card5, card8])
+        deck2 = Deck.new([card3, card4, card6, card7])
+
+
+        player1 = Player.new("Megan", deck1)
+        player2 = Player.new("Aurora", deck2)
+
+        turn = Turn.new(player1, player2)
+
+        expect(turn.type).to eq(:mutually_assured_destruction)
+
+        expect(turn.winner).to eq(nil)
```

Comment:
The project requirements indicate this should actually return "No Winner". I could see a lot of reasons why `nil` makes more sense, but technically this is not built to spec.

Reply:
Yeah. At the time, I wasn't entirely clear on whether what needed to be returned was a print of "No Winner" or "No Winner" itself, and I didn't know at the time that I could plug in print "No Winner" as a test, so I had a different part of my code handle printing out this scenario. I also recall running into problems when I tried to use the return value of this later while running the game.

I thought the pry commands and outputs were more of an example rather than needing to be followed literally.